plugins/renepay/pay.c

```diff
@@ -0,0 +1,1798 @@
+#include "config.h"
+#include <ccan/array_size/array_size.h>
+#include <ccan/cast/cast.h>
+#include <ccan/htable/htable_type.h>
+#include <ccan/tal/str/str.h>
+#include <common/bolt11.h>
+#include <common/bolt12_merkle.h>
+#include <common/gossmap.h>
+#include <common/json_param.h>
+#include <common/json_stream.h>
+#include <common/memleak.h>
+#include <common/pseudorand.h>
+#include <common/type_to_string.h>
+#include <errno.h>
+#include <plugins/renepay/debug.h>
+#include <plugins/renepay/pay.h>
+#include <plugins/renepay/pay_flow.h>
+#include <plugins/renepay/uncertainty_network.h>
+
+// TODO(eduardo): maybe there are too many debug_err and plugin_err and
+// plugin_log(...,LOG_BROKEN,...) that could be resolved with a command_fail
+
+#define INVALID_ID UINT64_MAX
+#define MAX(a,b) ((a)>(b)? (a) : (b))
+
+static struct pay_plugin the_pay_plugin;
+struct pay_plugin * const pay_plugin = &the_pay_plugin;
```

Comment:
Might be better to allocate this in init, using tal.  Then it can have natural tal children.

Reply:
When calling plugin `plugin_main` I am using `plugin_option` to load config files parameters into the data segment allocated `pay_plugin`. This can't be done if `pay_plugin` is allocated only after calling `init`.

I've seen other plugins that save config file options to separate global variables instead.
But I wanted to have them nicely packed inside the `pay_plugin` data structure, hence whenever I pass it around I can access my options all there in a single place.